test/integration/mesh/mesh_test.go

```diff
@@ -0,0 +1,166 @@
+package mesh_test
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework/utils"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/integration/mesh"
+
+	"github.com/aws/aws-sdk-go/aws"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Mesh", func() {
```

Comment:
Can we add a testcase for Mesh Delete scenario to validate the use case where we can't delete a mesh if there are appmesh resources under it? Also, to validate that a mesh delete will not automatically trigger delete of appmesh resources under it.Once we delete those resources mesh should be deleted.

-> Create a Mesh
-> Create a VN for ex
-> Delete Mesh -> should be stuck in Terminating state
-> Delete VN and check if Mesh is deleted as well

The above steps will help validate the mesh finalizer code flow..

Reply:
Thanks for the review. All valid scenarios. Yes, I plan to add these scenarios in a follow up. This is just the baseline to get started on the integration tests